packages/lit-dev-content/site/docs/v3/data/task.md

```diff
@@ -0,0 +1,440 @@
+---
+title: Async Tasks
+eleventyNavigation:
+  key: Async Tasks
+  parent: Managing Data
+  order: 2
+---
+
+## Overview
+
+Sometimes a component needs to render data that is only available _asynchronously_. Such data might be fetched from a server, a database, or in general retrieved or computed from an async API.
+
+While Lit's reactive update lifecycle is batched and asynchronous, Lit templates always render _synchronously_. The data used in a template must be readable at the time of rendering. To render async data in a Lit component, you must wait for the data to be ready, store it so that it's readable, then trigger a new render which can use the data synchronously. Considerations must often be made on what to render while the data is being fetched, or when the data fetch fails as well.
+
+The `@lit/task` package provides a `Task` reactive controller to help manage this async data workflow.
+
+`Task` is a controller that takes an async task function and runs it either manually or automatically when its arguments change. Task stores the result of the task function and updates the host element when the task function completes so the result can be used in rendering.
+
+### Example
+
+This is an example of using `Task` to call an HTTP API via [`fetch()`](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API). The API is called whenever the `productId` parameter changes, and the component renders a loading message when the data is being fetched.
+
+{% switchable-sample %}
+
+```ts
+import {Task} from '@lit-labs/task';
+
+class MyElement extends LitElement {
+  @property() productId?: string;
+
+  private _productTask = new Task(this, {
+    task: async ([productId], {signal}) => {
+      const response = await fetch(`http://example.com/product/${productId}`, {signal});
```

Comment:
Would it be useful if examples worked in the lit.dev playground without too many changes?

I wanted to try this out, but it required me fixing the URL to `https:` and adding `mode: "no-cors",` to the fetch options. Then the URL always seems to return 404 (which I think makes sense because it's an example URL).